### DISCUSSION TITLE
22 minutes per query

### DISCUSSION THREAD

**Discussion by VGPS** [imartinez/privateGPT#734] [Q&A]
Hi - love the app! I'm learning a lot through using it. Unfortunately each query takes around 20 minutes using the state_of_the_union document. My CPU usage hovers around 98% during the query. I clearly need a more powerful PC. What PC configuration would anyone recommend to speed up the process? Would off-loading the db and itsfiles and the model to a clean remote drive make any difference? Keep up the good work!

**Comment by egorgutorovv** [comment]
Hi! Write here your configuration pls

**Reply by VGPS** [reply]
processor: Intel(R) Core(TM) i5-3340M CPU @ 2.70GHz   2.70 GHz

Installed RAM: 16 GB

Hard drive free space: 25G

Thanks!

**Reply by egorgutorovv** [reply]
I am sure your proposed steps for speeding up performance won't affect much.  I believe that **so far** such LLM projects (for any production purposes) should be launched only with powerfull GPU. An option for CPU is just for playing around at that period of time.  

My config looks like that:
Intel(R) Core(TM) i7-6700HQ CPU @ 2.60GHz   2.60 GHz
16gb RAM and HDD with free 800GB. 

For me it took about 10 minutes to handle with query "Who is Putin?" using the state_of_the_union document. I use model ggml-vic13b-q5_1.  While query processor is working average 60%.

**Reply by VGPS** [reply]
Thanks - I'll try that model and see if it makes any difference. Any recommendations for a PC with a GPU? I'm new to this and have never used a GPU that I'm aware of.

**Reply by egorgutorovv** [reply]
For me GPU is also unknown domain) I see other guys discussing the topic in https://github.com/imartinez/privateGPT/discussions/217#discussion-5202797 Maybe it's gonna prompt you way.
